test/server/cards/02_SHD/units/ChainCodeCollector.spec.ts

```diff
@@ -0,0 +1,39 @@
+describe('Chain Code Collector', function () {
+    integration(function (contextRef) {
+        describe('Chain Code Collector\'s on attack ability', function () {
+            it('should give -4/-0 on unit who have bounty', function () {
+                const { context } = contextRef;
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['chain-code-collector'],
```

Comment:
I don't think there's any major concern but just for an integration test it might be good to start the unit in hand so we can confirm the ability works right with the Ambush attack